io/plugins/eu.esdihumboldt.hale.io.csv/src/eu/esdihumboldt/hale/io/csv/reader/internal/CSVInstanceCollection.java

```diff
@@ -85,8 +85,21 @@ private void proceedToNext() {
 
 			// initialize reader if necessary
 			if (csvReader == null) {
-				boolean skipFirst = reader.getParameter(CommonSchemaConstants.PARAM_SKIP_FIRST_LINE)
-						.as(Boolean.class, false);
+
+				int skipN = 0;
+				Boolean skipType = reader.getParameter(CommonSchemaConstants.PARAM_SKIP_N_LINES)
+						.as(boolean.class);
+
+				if (skipType == null) {
+					skipN = reader.getParameter(CommonSchemaConstants.PARAM_SKIP_N_LINES)
+							.as(Integer.class, 0);
+				}
+				else if (skipType == true) {
+					skipN = 1;
+				}
+				else if (skipType == false) {
```

Comment:
Here you don't need `else if` just `else` is enough as the boolean parameter skipType can have only 3 values:
- null
- true
- false
so for the third condition you don't have to explicitly do a check for false.